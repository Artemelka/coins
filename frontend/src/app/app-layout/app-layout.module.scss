.root {
  height: 100vh;
  width: 100vw;
  display: flex;
  flex-direction: column;
  align-items: stretch;
  justify-content: stretch;
}

.header {
  width: 100%;
  padding: 15px 30px;
  background-color: var(--baseBackgroundColor);
}

.main {
  max-width: 100%;
  min-width: 100%;
  flex-grow: 2;
  max-height: 100%;
  overflow-y: scroll;
}
