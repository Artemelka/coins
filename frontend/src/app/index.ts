export { App } from './app';
export type { AppStore, BaseAction } from './types';