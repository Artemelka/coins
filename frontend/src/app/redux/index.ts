export { ReduxProvider } from './_components';