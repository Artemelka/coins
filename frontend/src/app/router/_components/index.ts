export { AppRouterProvider } from './app-router-provider';
export { AppRouterSwitch } from './app-router-switch';