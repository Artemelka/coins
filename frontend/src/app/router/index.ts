export {
  AppRouterProvider,
  AppRouterSwitch,
} from './_components';
export { appRouterMiddleware } from './middleware';
export { ROUTER_REDUCER_KEY } from './constants';
export { routerReducer } from './reducer';
export * from './selectors';
export type { AppRouterProps, RouterStore } from './types';