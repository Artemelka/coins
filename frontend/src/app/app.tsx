import React, { FC, memo, useEffect } from 'react';
import { ReduxProvider } from './redux';
import { AppRouterProvider } from './router';
import { AppLayout } from './app-layout';
import { PAGES_ROUTES_CONFIG } from '@/pages';
import { bootstrapApp } from './_utils/bootstrap-app';

export const AppContainer: FC = () => {
  useEffect(() => {
    bootstrapApp();
  }, []);

  return (
    <ReduxProvider>
      <AppRouterProvider>
        <AppLayout routes={PAGES_ROUTES_CONFIG}/>
      </AppRouterProvider>
    </ReduxProvider>
  );
}

export const App = memo(AppContainer);
