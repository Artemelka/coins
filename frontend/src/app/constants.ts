import { ROUTER_REDUCER_KEY } from './router';

export const APP_STORE_KEYS = {
  ROUTER: ROUTER_REDUCER_KEY
} as const;