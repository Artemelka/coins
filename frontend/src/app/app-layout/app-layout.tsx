import React, { FC, memo } from 'react';
import { AppRouterSwitch, Route, AppRouterProps } from '@/app/router';
import { ThemeSwitcher } from '@/components';
import styles from './app-layout.module.scss';

type AppLayoutProps = {
  routes: Array<AppRouterProps>;
};

export const AppLayoutComponent: FC<AppLayoutProps> = ({ routes }) => {
  return (
      <div className={styles.root}>
        <header className={styles.header}>
            <ThemeSwitcher/>
        </header>
        <main className={styles.main}>
          <AppRouterSwitch pathToRedirect="/coins">
            {routes.map(props => <Route key={props.path} {...props} />)}
          </AppRouterSwitch>
        </main>
      </div>
  );
};

export const AppLayout = memo(AppLayoutComponent);