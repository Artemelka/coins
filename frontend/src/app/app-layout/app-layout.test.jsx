import React from 'react';
import { shallow } from 'enzyme';
import { AppLayoutComponent } from './app-layout';
import { AppRouterSwitch, Route } from '@/app/router';
import { PAGES_ROUTES_CONFIG } from '@/pages';

describe('AppLayoutComponent', () => {
    test('render', () => {
        const component = shallow(<AppLayoutComponent routes={PAGES_ROUTES_CONFIG}/>);

        expect(component).toMatchSnapshot();
    });

    test('tag header should be', () => {
        const component = shallow(<AppLayoutComponent routes={PAGES_ROUTES_CONFIG}/>);

        expect(component.find('.header').type()).toBe('header');
    });

    test('tag main should be', () => {
        const component = shallow(<AppLayoutComponent routes={PAGES_ROUTES_CONFIG}/>);

        expect(component.find('.main').type()).toBe('main');
    });

    test('AppRouterSwitch should have correct pathToRedirect', () => {
        const component = shallow(<AppLayoutComponent routes={PAGES_ROUTES_CONFIG}/>);

        expect(component.find(AppRouterSwitch).props().pathToRedirect).toBe(PAGES_ROUTES_CONFIG[0].path);
    });

    test('first Route should have correct path', () => {
        const component = shallow(<AppLayoutComponent routes={PAGES_ROUTES_CONFIG}/>);

        expect(component.find(Route).get(0).props.path).toBe(PAGES_ROUTES_CONFIG[0].path);
    });
});