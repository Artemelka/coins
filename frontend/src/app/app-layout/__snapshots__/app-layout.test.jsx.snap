// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppLayoutComponent render 1`] = `
<div
  className="root"
>
  <header
    className="header"
  >
    <Memo(ThemeSwitcherComponent) />
  </header>
  <main
    className="main"
  >
    <Memo(AppRouterSwitchComponent)
      pathToRedirect="/coins"
    >
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "compare": null,
            "type": [Function],
          }
        }
        key="/coins"
        path="/coins"
      />
    </Memo(AppRouterSwitchComponent)>
  </main>
</div>
`;
